### DISCUSSION TITLE
License and Attribution

### DISCUSSION THREAD

**Discussion by controlfreakstudio** [funmusicplace/mirlo#499] [Ideas]
Part of fostering good community vibes for respecting artists' works, consent, agency and being open and clear about inviting other uses or instigating good communication about amiable partnerships for sharing and using works is having good and clear license and attribution listed .

Can you please make this an artist editing option if not already and give it a prominent visual location to travel per track/release?

Due to collaborations or intended use cases and many variables artist license preference can vary per track/release.

**Comment by simonv3** [comment]
@controlfreakstudio that's a great idea! So you think this should be something an artist selects for each track? I'm also wondering what you think the default should be for any artist who doesn't make a stance on it? I guess it would just be normal copyright?

**Reply by controlfreakstudio** [reply]
Very much so.  Sometimes a collab track might have a different. SOmetimes a specific track on a release is wanting to be made avaialbe for more open types of re-use.

Understand that if it's an open field it might be confusing for artists that haven't chosen yet, aren't sure, or don't care? But defaulting to most strict Copyright is probably the best etiquette. Maybe some other options in a drop or at least an "Other" open field other than default copyright for those that know exactly what they want.

usually good to include a little primer somewhere that links some artist friendly info on basic choices around copyright or a few popular more open options with the clear intent that it's not legal advice nor an endorsement, just helping artists know that so long as they have not signed over rights in some other manner it's their choice to make and hopefully it's an empowered one and your platform will respect passing that on to the listener etc.

**Comment by simonv3** [comment]
Note that some artists put their license info in the credits: https://mirlo.space/dagure/release/mrrr.-12-(live)

**Comment by Cwpute** [comment]
Please when this is implemented: link every licence to their original material/website ! So that when browsing through an album you can actually have a short, curious look at what it's all about ! And when users provide a custom licence, allow them and entice them into adding a link to that licence's full text/website as well !
Too many times i've seen licences slapped on stuff i was browsing, and i couldn't care less about it. A link to something tangible, that you can read for yourself (https://creativecommons.org/licenses/by-sa/2.0/fr/deed.fr), or which gets explained to you by the creators of that licence (http://wtfpdl.net/index.html) is always so much better i believe. It also empowers the people about content ownership and use because the you provide them the resources to learn about it :)

It's something i wish other places which host community-content would do !

**Comment by lukaprincic** [comment]
could licences be actually hosted on mirlo in git(hub/lab/forgejo) style - and added to a zip in a download (when downloading multiple files in a zip)? you know when you create a project on one of web-git platforms you get a dropdown menu of licences that should be included in the repo. the reason to use that with music is that for some of us CC licences are not ok - for example there's a better licence that states clearly that commercial use is allowed only to coops and worker-owned enterprises but not to classical corporations or for-profit few-people-owned companies - i.e. Peer Production Licence.

**Reply by simonv3** [reply]
I think this is handled by letting users add new types of license to tracks, rather than presetting a list right?

**Reply by lukaprincic** [reply]
> I think this is handled by letting users add new types of license to tracks, rather than presetting a list right?

not sure how is this implemented? it would be great to allow the uploader **choose** a licence, not **add** a licence for every upload. also, if there's a database of licences, it would be great to be able to browse/search/list music released under a specific licence on mirlo.

i think how this is handled on git web platforms is great - for example on codeberg, when creating a new repository:

![image](https://github.com/user-attachments/assets/9f07ce9d-6ec4-400f-9b15-2b7b55cc50bf)

ideally there would be moderated/curated collection of licences that would be offered in such a dropdown menu for the uploader. but perhaps it would be very easy to suggest an addition to such menu/list and/or add a custom one.

**Reply by simonv3** [reply]
https://github.com/user-attachments/assets/f0bd1f80-5006-4045-acbb-08341617ab1e



This is how it's currently implemented. If you edit a track you can choose the license. ATM anyone can add licenses and we can re-evaluate that if it becomes a problem. I'm happy to also occasionally tweak the text in the DB to keep it consistent

**Comment by simonv3** [comment]
This does a very basic implementation for this on a per-track basis https://github.com/funmusicplace/mirlo/pull/805

Edit: I'm going to close this as the original suggestion has been implemented. if people think there need to be changes to the actual implementation then we can discuss that in a issue
